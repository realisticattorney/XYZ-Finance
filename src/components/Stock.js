/* eslint-disable */
import React from 'react';
import PropTypes from 'prop-types';
import graph from '../assets/graph.png';
/* eslint-disable array-callback-return */
function Stock({ stock }) {
  const {
    logo, symbol, price, exchange,
  } = stock;

  function importAll(r) {
    const images = {};
    r.keys().map((item) => {
      images[item.replace('./', '')] = r(item);
    });
    return images;
  }

  const images = importAll(require.context('../assets', false, /\.(png|jpe?g|svg)$/));

  return (
    <div className="book">
      {/* <p className="category">{id}</p> */}
      <p className="title">{price}</p>
      <p className="author">{symbol}</p>
      <p className="category">{exchange}</p>
      <img src={images[logo].default} alt={symbol} className="logos" />
      <div className="rect">
        <div className="circle">
          <img src={graph} alt="Logo" />
        </div>
      </div>
      <div className="hr" />
      <div className="update">
        <p className="chapter">Current Chapter</p>
        <p className="intro">Introduction</p>
        <button type="button">Update Progress</button>
      </div>
    </div>
  );
}

Stock.propTypes = {
  stock: PropTypes.shape({
    price: PropTypes.string.isRequired,
    exchange: PropTypes.string.isRequired,
    symbol: PropTypes.string.isRequired,
    logo: PropTypes.string.isRequired,
  }).isRequired,
};
export default Stock;
