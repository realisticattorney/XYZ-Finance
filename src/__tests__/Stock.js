import React from 'react';
import renderer from 'react-test-renderer';
import Stock from '../components/Stock';

it('renders Stock component', () => {
  const stock = {
    symbol: 'AAPL',
    companyName: 'Apple Inc.',
  };
  const stockSnap = renderer.create(<Stock stock={stock} />);
  expect(stockSnap).toMatchSnapshot();
});
