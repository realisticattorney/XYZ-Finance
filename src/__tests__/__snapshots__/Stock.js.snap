// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Stock component 1`] = `
<a
  href="/stock/AAPL"
>
  <div
    className="book"
  >
    <div
      className="left-side"
    >
      <p
        className="category"
      >
        Apple Inc.
      </p>
      <p
        className="author"
      >
        AAPL
      </p>
    </div>
    <div
      className="right-side"
    >
      <p
        className="title"
      />
    </div>
  </div>
</a>
`;
